osquery/utils/conversions/windows/strings.cpp

```diff
@@ -79,4 +84,48 @@ std::string bstrToString(const BSTR src) {
   return wstringToString(static_cast<const wchar_t*>(src));
 }
 
+LONGLONG cimDatetimeToUnixtime(const std::string& src) {
+  // First init the SWbemDateTime class
+  ISWbemDateTime* pCimDateTime = nullptr;
+  auto hres = CoCreateInstance(CLSID_SWbemDateTime,
+                               nullptr,
+                               CLSCTX_INPROC_SERVER,
+                               IID_PPV_ARGS(&pCimDateTime));
+  if (!SUCCEEDED(hres)) {
+    LOG(WARNING) << "Failed to init CoCreateInstance with " << hres;
+    return -1;
+  }
+
+  // Then load up our CIM Datetime string into said class
+  auto bSrcStr = SysAllocString(stringToWstring(src.c_str()).c_str());
+  auto const bSrcStrManager =
+      scope_guard::create([&bSrcStr]() { SysFreeString(bSrcStr); });
+  hres = pCimDateTime->put_Value(bSrcStr);
+  if (!SUCCEEDED(hres)) {
+    LOG(WARNING) << "Failed to init CimDateTime with " << hres;
+    return -1;
+  }
+
+  // Convert this CIM Datetime to a FILETIME BSTR
+  BSTR bstrFileTime{L""};
+  auto const bstrFileTimeManager =
+      scope_guard::create([&bstrFileTime]() { SysFreeString(bstrFileTime); });
+  // VARIANT_FALSE means we fetch the time in UTC
+  hres = pCimDateTime->GetFileTime(VARIANT_FALSE, &bstrFileTime);
+  if (!SUCCEEDED(hres)) {
+    LOG(WARNING) << "Failed to convert CimDateTime to FILETIME with " << hres;
+    return -1;
+  }
+
+  LARGE_INTEGER intStore;
+  intStore.QuadPart = _wtoi64(bstrFileTime);
+  FILETIME timeStore;
+
+  timeStore.dwLowDateTime = intStore.LowPart;
+  timeStore.dwHighDateTime = intStore.HighPart;
+
+  // And finally convert this to a Unix epoch timestamp
+  return filetimeToUnixtime(timeStore);
```

Comment:
I recommend removing this function from `fileops.cpp` and bringing it into the `./utils/conversation` (where ever is most appropriate) since it is simple and innocuous. We should not make a conversions library like this one depend on the filesystem library.

This will allow you to remove the CMake dependency above, which will fix the linking error.

Reply:
Done. I took a stab at this, I created a new header/implementation combo for `windows_time` under `util/conversions`. Happy to go with something else if folks feel strongly, but this seemed appropriate to me, as the only other conversions we seem to have look to largely be string based.